packages/appkit/src/adapters/ChainAdapterBlueprint.ts

```diff
@@ -0,0 +1,426 @@
+   * @returns {Promise<AdapterBlueprint.GetBalanceResult>} Balance result
+   */
+  public abstract getBalance(
+    params: AdapterBlueprint.GetBalanceParams
+  ): Promise<AdapterBlueprint.GetBalanceResult>
+
+  /**
+   * Gets the profile for a given address and chain ID.
+   * @param {AdapterBlueprint.GetProfileParams} params - Profile retrieval parameters
+   * @returns {Promise<AdapterBlueprint.GetProfileResult>} Profile result
+   */
+  public abstract getProfile(
+    params: AdapterBlueprint.GetProfileParams
+  ): Promise<AdapterBlueprint.GetProfileResult>
+
+  /**
+   * Synchronizes the connectors with the given options and AppKit instance.
+   * @param {AppKitOptions} [options] - Optional AppKit options
+   * @param {AppKit} [appKit] - Optional AppKit instance
+   */
+  public abstract syncConnectors(options?: AppKitOptions, appKit?: AppKit): void
+
+  /**
+   * Synchronizes the connection with the given parameters.
+   * @param {AdapterBlueprint.SyncConnectionParams} params - Synchronization parameters
+   * @returns {Promise<AdapterBlueprint.ConnectResult>} Connection result
+   */
+  public abstract syncConnection(
+    params: AdapterBlueprint.SyncConnectionParams
+  ): Promise<AdapterBlueprint.ConnectResult>
+
+  /**
+   * Signs a message with the connected wallet.
+   * @param {AdapterBlueprint.SignMessageParams} params - Parameters including message to sign, address, and optional provider
+   * @returns {Promise<AdapterBlueprint.SignMessageResult>} Object containing the signature
+   */
+  public abstract signMessage(
+    params: AdapterBlueprint.SignMessageParams
+  ): Promise<AdapterBlueprint.SignMessageResult>
+
+  /**
+   * Estimates gas for a transaction.
+   * @param {AdapterBlueprint.EstimateGasTransactionArgs} params - Parameters including address, to, data, and optional provider
+   * @returns {Promise<AdapterBlueprint.EstimateGasTransactionResult>} Object containing the gas estimate
+   */
+  public abstract estimateGas(
+    params: AdapterBlueprint.EstimateGasTransactionArgs
+  ): Promise<AdapterBlueprint.EstimateGasTransactionResult>
+
+  /**
+   * Sends a transaction.
+   * @param {AdapterBlueprint.SendTransactionParams} params - Parameters including address, to, data, value, gasPrice, gas, and optional provider
+   * @returns {Promise<AdapterBlueprint.SendTransactionResult>} Object containing the transaction hash
+   */
+  public abstract sendTransaction(
+    params: AdapterBlueprint.SendTransactionParams
+  ): Promise<AdapterBlueprint.SendTransactionResult>
+
+  /**
+   * Writes a contract transaction.
+   * @param {AdapterBlueprint.WriteContractParams} params - Parameters including receiver address, token amount, token address, from address, method, and ABI
+   * @returns {Promise<AdapterBlueprint.WriteContractResult>} Object containing the transaction hash
+   */
+  public abstract writeContract(
+    params: AdapterBlueprint.WriteContractParams
+  ): Promise<AdapterBlueprint.WriteContractResult>

```

Comment:
Following up the previous comment, I think `writeContract`, `getEnsAddress`, `parseUnits`, `formatUnits` and so on should be part of the Connector abstraction and not part of the ChainAdapterBlueprint. I mean this because those methods are specifics of EVM networks and should not be needed in every chain.

We could have interfaces like `EVMConnector` or `SolanaConnector` or `BitcoinConnector` extending `ChainAdapterConnector` and implementing the specific methods of each chain, this will be later the `connector` present in this Blueprint prototype and what we expose for our users. With these interfaces we can create classes like `class ExtensionWallet extends EVMConnector` and this specific class would handle extensions to do what the connector should do. This is what would create abstraction between Wallet Connect, Extension Wallet or Auth and also would really help to not need to keep using type casting everywhere.